### PR TITLE
Update `hydrotools.metrics` to newer Python versions

### PR DESCRIPTION
**Pull request by jarq6c**
This PR updates the `hydrotools.metrics` package to test on Python versions 3.9, 3.10, and 3.11. It also transitions the package from using a `setup.cfg` to just using `pyproject.toml`.

## Additions

-

## Removals

- `setup.cfg`

## Changes

- `pyproject.toml`
- Had to switch some docstrings to raw strings due to embeded LaTeX.

## Testing

1. No changes.

## Notes

- Last version under old DOC license.
- Related to #259 and #260 

## Todos

-

## Checklist

- [x] PR has an informative and human-readable title
- [ x PR is well outlined and documented. See [#12](https://github.com/jarq6c/evaluation_tools/pull/12) for an example
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows project standards (see [CONTRIBUTING.md](../CONTRIBUTING.md))
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output) using [numpy docstring](https://numpydoc.readthedocs.io/en/latest/format.html) formatting
- [ ] Placeholder code is flagged / future todos are captured in comments
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by jarq6c** [comment]
> All tests past. Going to do a test deployment to see if classifiers/hooks get picked up correctly without `setup.cfg`.

Deployment looked good. Just needed to revert 7519e7d5dde06f17549b2c487cc7a7b281c152c7

We'll adopt the new license once the updated GH policy is released.

File: python/metrics/pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,41 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "wheel",
-    "pip"
-]
+requires = ["setuptools>=61"]
```
Comment (aaraney) [inline python/metrics/pyproject.toml]: Lets bump this to `"setuptools>=70.0"`. See [CVE-2024-6345](https://nvd.nist.gov/vuln/detail/CVE-2024-6345) for motivation.

File: python/metrics/src/hydrotools/metrics/metrics.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def mean_error(
     power: float = 1.0,
     root: bool = False
     ) -> float:
-    """Compute the mean error or deviation. Default is Mean Absolute Error. The mean error 
+    r"""Compute the mean error or deviation. Default is Mean Absolute Error. The mean error
```
Comment (aaraney) [inline python/metrics/src/hydrotools/metrics/metrics.py:47]: This is the first time i've seen a docstring defined as a raw string literal. I assume this avoids character escaping headaches in the latex expressions?

**Comment by jarq6c** [comment]
c9b0932a2628514b7b12490bbe00ae6fde4b6fc2 related to: https://nvd.nist.gov/vuln/detail/CVE-2024-6345

Thanks @aaraney !

**Review by aaraney — APPROVED** [review]
Looks great to me! Thanks, @jarq6c!

File: python/metrics/pyproject.toml (hunk 1 above)
Comment (jarq6c) [reply python/metrics/pyproject.toml]: Good catch!

File: python/metrics/src/hydrotools/metrics/metrics.py (hunk 2 above)
Comment (jarq6c) [reply python/metrics/src/hydrotools/metrics/metrics.py:47]: Yeah, the expressions were raising DeprecationWarnings and this was the easiest solution.

**Comment by jarq6c** [comment]
Deployed: https://pypi.org/project/hydrotools.metrics/1.3.5/

**Pull request merged**